ai-development-guidelines-v2.md

```diff
@@ -0,0 +1,375 @@
+# AI-Assisted Development: Human-in-the-Loop Guidelines
+
+## Executive Summary
+
+This document establishes comprehensive guidelines for integrating AI code assistants (claude.code, GitHub Copilot, etc.) into software development workflows while maintaining critical human oversight and quality control.
+
+## 1. Development Workflow Requirements
+
+### 1.1 Pre-Development Phase
+
+**AI Usage Declaration**
+- Engineers must declare AI assistant usage in commit messages using standardized tags:
+  - `[AI-ASSISTED]` - AI helped with implementation
+  - `[AI-GENERATED]` - Majority of code generated by AI
+  - `[AI-REVIEWED]` - AI used for code review suggestions
+
+**Human Validation Checkpoints**
+- **Requirement Analysis**: Human must validate AI's understanding of requirements before implementation
+- **Architecture Review**: Senior engineer approval required for AI-suggested architectural decisions
+- **Security Impact Assessment**: Human security review mandatory for any AI-generated code touching:
+  - Authentication/authorization
+  - Data validation
+  - External APIs
+  - Database operations
+
+### 1.2 Development Phase Guardrails
+
+**Code Generation Rules**
+1. **Maximum AI Block Size**: No single AI-generated code block exceeds 50 lines without human review
+2. **Critical Path Restriction**: AI cannot generate code for:
+   - Payment processing
+   - User authentication
+   - Data encryption/decryption
+   - Database migration scripts
+   - Production deployment scripts
```

Comment:
I agree that we should be _very_ strict on accepting these kind of changes, we have no way to enforce this.

I would suggest the guideline being that these kind of changes require strict human review.